src/sql/azdata.proposed.d.ts

```diff
@@ -2006,6 +2006,127 @@ declare module 'azdata' {
 		}
 	}
 
+	export type ChartType = 'bar' | 'bubble' | 'doughnut' | 'horizontalBar' | 'line' | 'pie' | 'polarArea' | 'radar' | 'scatter';
+
+	export interface ModelBuilder {
+		chart<T extends ChartOptions>(): ComponentBuilder<ChartComponent<T>, ChartComponentProperties<T>>;
+	}
+
+	export interface ChartComponentProperties<T extends ChartOptions> extends ComponentProperties {
+		chartType: ChartType;
+		data: ChartData;
+		options?: T;
+	}
+
+	export interface BubbleChartPoint {
+		x: number;
+		y: number;
+		r: number;
+	}
+
+	export interface ScatterChartPoint {
+		x: number;
+		y: number;
+	}
+
+	export interface ChartData {
+		label?: string;
+		dataset: number[] | BubbleChartPoint[] | ScatterChartPoint[];
+		labels?: string[];
+		colors?: string[];
```

Comment:
What happens if the lengths of these arrays don't match?  If there's a failure, then perhaps the modelview component should improve the API (e.g. with `type ChartDataPoint = { value: number | BubbleChartPoint | ScatterChartPoint; label?: string; color? string}`